meilisearch/models/task.py

```diff
@@ -1,32 +1,29 @@
 from __future__ import annotations
 
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Union
 
 from camel_converter.pydantic_base import CamelBase
 
 
 class Task(CamelBase):
     uid: str
-    index_uid: str
+    index_uid: Union[str, None]
     status: str
     type: str
-    details: Optional[Dict[str, Any]]
+    details: Dict[str, Any]
+    error: Union[str, None]
```

Comment:
As for before v0.30.0 error is still an optional field.  In v0.30 it becomes an Union